### PR TITLE
Feature/192 update server applications

### PR DESCRIPTION
**Pull request by lrebscher**
**Manual testing:**
* Run harness-deployment and check if images build with success
* Install helm chart in minikube and check if applications start and show no error in logs

Closes #194

### CODE REVIEW

**Review by filippomc — APPROVED** [review]
Looks good and safe to merge

File: applications/common/server/Dockerfile [hunk 1]
```diff
@@ -1,24 +1,21 @@
...
 ARG TAG=latest
-FROM ${REGISTRY}cloudharness-base:${TAG}
+FROM ${REGISTRY}cloudharness-flask:${TAG}
+
+ENV MODULE_NAME=common
+
+ENV WORKERS=2
+ENV PORT=8080
 
 RUN apk update && apk add postgresql-dev gcc python3-dev musl-dev
```
Comment (zsinnema) [inline applications/common/server/Dockerfile:10]: musl-dev, python3-dev, gcc can be removed, they are part of the cloudharness-base image

File: applications/common/server/Dockerfile [hunk 2]
```diff
@@ -1,24 +1,21 @@
...
+# TODO: @Sinnema can this be removed?
 #RUN apk add psycopg2 libpq-dev python-dev
```
Comment (zsinnema) [inline applications/common/server/Dockerfile:12]: can be removed

File: applications/common/server/common/__main__.py [hunk 3]
```diff
@@ -1,23 +1,45 @@
...
 from common import encoder
 
-def main():
-    app = connexion.App(__name__, specification_dir='./openapi/')
-    app.app.config.from_object(os.environ['APP_SETTINGS'])
-    app.app.json_encoder = encoder.JSONEncoder
-    app.add_api('openapi.yaml',
-                arguments={'title': 'CH service API'},
-                pythonic_params=True)
+connexion_app = connexion.App(__name__, specification_dir='./openapi/')
```
Comment (zsinnema) [inline applications/common/server/common/__main__.py:12]: please wrap the whole initialisation into a CH function, looks to me that this is everytime the same

please use cloudharness.log and remove the unicorn logger stuff (2 weeks ago I formatted the default CH logger)

File: applications/common/server/common/__main__.py [hunk 4]
```diff
@@ -1,23 +1,45 @@
...
+        # init_app can be defined to add behaviours to the wsgi app
+        from common import init_app
+
+        init_app(app)
+    except ImportError:
+        pass
+
+
+def main():
+    connexion_app.run(port=8080)
```
Comment (zsinnema) [inline applications/common/server/common/__main__.py:41]: may be better to remove the hardcoded 8080 port (set it via the FLASK_RUN_PORT or the command line parameter)

File: applications/common/server/common/__main__.py [hunk 5]
```diff
@@ -1,23 +1,45 @@
...
+    app.logger.setLevel(gunicorn_logger.level)
+
     from .repository.db import open_db
+
     open_db(app)
-    cors = CORS(app.app, resources={r"/api/*": {"origins": "*"}})
-    app.run(port=8080)
+    cors = CORS(app, resources={r"/api/*": {"origins": "*"}})
+    connexion_app.add_api('openapi.yaml',
+                          arguments={'title': 'CH service API'},
```
Comment (zsinnema) [inline applications/common/server/common/__main__.py:28]: the title would then be configurable (a parameter of the init routine)

File: applications/volumemanager/server/volumemanager/__init__.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+from flask import Flask, send_from_directory
+
+
+def init_app(app: Flask):
```
Comment (zsinnema) [inline applications/volumemanager/server/volumemanager/__init__.py:4]: why add the routers if there are no files to server

File: applications/volumemanager/server/volumemanager/__main__.py [hunk 7]
```diff
@@ -1,18 +1,35 @@
 #!/usr/bin/env python3
 
-import connexion
+import logging
```
Comment (zsinnema) [inline applications/volumemanager/server/volumemanager/__main__.py:3]: use the ch logging

File: applications/volumemanager/server/volumemanager/__main__.py [hunk 8]
```diff
@@ -1,18 +1,35 @@
...
 
+import connexion
 from volumemanager import encoder
 
+# setup connection app
+connexion_app = connexion.App(__name__, specification_dir="./openapi/", debug=True)
```
Comment (zsinnema) [inline applications/volumemanager/server/volumemanager/__main__.py:9]: same here, may be we can create a default initializer

File: applications/volumemanager/server/volumemanager/__main__.py [hunk 9]
```diff
@@ -1,18 +1,35 @@
...
+
 
 def main():
-    app = connexion.App(__name__, specification_dir='./openapi/')
-    app.app.json_encoder = encoder.JSONEncoder
-    app.add_api('openapi.yaml',
-                arguments={'title': 'Volumes manager API'},
-                pythonic_params=True)
-    app.run(port=8080)
+    connexion_app.debug = True
```
Comment (zsinnema) [inline applications/volumemanager/server/volumemanager/__main__.py:30]: debug?

File: applications/volumemanager/server/volumemanager/__main__.py [hunk 10]
```diff
@@ -1,18 +1,35 @@
...
+    connexion_app.run(port=5000)
```
Comment (zsinnema) [inline applications/volumemanager/server/volumemanager/__main__.py:31]: remove port

File: applications/workflows/server/workflows_api/__init__.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
+from flask import Flask, send_from_directory
+
+
+def init_app(app: Flask):
```
Comment (zsinnema) [inline applications/workflows/server/workflows_api/__init__.py:4]: why routers if there are no files?

File: applications/workflows/server/workflows_api/__main__.py [hunk 12]
```diff
@@ -1,18 +1,35 @@
 #!/usr/bin/env python3
 
-import connexion
+import logging
```
Comment (zsinnema) [inline applications/workflows/server/workflows_api/__main__.py:3]: ch logging

File: applications/workflows/server/workflows_api/__main__.py [hunk 13]
```diff
@@ -1,18 +1,35 @@
...
 
+import connexion
 from workflows_api import encoder
 
+# setup connection app
+connexion_app = connexion.App(__name__, specification_dir="./openapi/", debug=True)
```
Comment (zsinnema) [inline applications/workflows/server/workflows_api/__main__.py:9]: init function?

File: applications/workflows/server/workflows_api/__main__.py [hunk 14]
```diff
@@ -1,18 +1,35 @@
...
+
 
 def main():
-    app = connexion.App(__name__, specification_dir='./openapi/')
-    app.app.json_encoder = encoder.JSONEncoder
-    app.add_api('openapi.yaml',
-                arguments={'title': 'Workflows API'},
-                pythonic_params=True)
-    app.run(port=8080)
+    connexion_app.debug = True
```
Comment (zsinnema) [inline applications/workflows/server/workflows_api/__main__.py:30]: debug?

File: applications/workflows/server/workflows_api/__main__.py [hunk 15]
```diff
@@ -1,18 +1,35 @@
...
+    connexion_app.run(port=5000)
```
Comment (zsinnema) [inline applications/workflows/server/workflows_api/__main__.py:31]: remove port

**Review by zsinnema — CHANGES_REQUESTED** [review]

**Comment by filippomc** [comment]
@lrebscher most of the requests are coming from the default application templates. I can do the improvements starting from there so to keep this future proof

**Comment by zsinnema** [comment]
> @lrebscher most of the requests are coming from the default application templates. I can do the improvements starting from there so to keep this future proof

sounds great!

**Comment by filippomc** [comment]
Closing as merged with #199 in #229

**Pull request closed without merging**